miio/yeelight_dual_switch.py

```diff
@@ -0,0 +1,254 @@
+    def switch_2_state(self) -> bool:
+        return bool(self.data["switch_2_state"])
+
+    @property
+    def switch_2_default_state(self) -> bool:
+        return bool(self.data["switch_2_default_state"])
+
+    @property
+    def switch_2_off_delay(self) -> int:
+        return self.data["switch_2_off_delay"]
+
+    @property
+    def interlock(self) -> bool:
+        return bool(self.data["interlock"])
+
+    @property
+    def flex_mode(self) -> int:
+        return self.data["flex_mode"]
+
+    @property
+    def rc_list(self) -> str:
+        return self.data["rc_list"]
+
+    def __repr__(self) -> str:
+        s = (
+            "<YeelightDualControlModuleStatus "
+            "switch_1_state=%s, "
+            "switch_1_default_state=%s, "
+            "switch_1_off_delay=%s, "
+            "switch_2_state=%s, "
+            "switch_2_default_state=%s, "
+            "switch_2_off_delay=%s, "
+            "interlock=%s, "
+            "flex_mode=%s, "
+            "rc_list=%s>"
+            % (
+                self.switch_1_state,
+                self.switch_1_default_state,
+                self.switch_1_off_delay,
+                self.switch_2_state,
+                self.switch_2_default_state,
+                self.switch_2_off_delay,
+                self.interlock,
+                self.flex_mode,
+                self.rc_list,
+            )
+        )
+        return s
+
+
+class YeelightDualControlModule(MiotDevice):
+    """Main class representing the air humidifier which uses MIoT protocol."""
+
+    def __init__(
+        self,
+        ip: str = None,
+        token: str = None,
+        start_id: int = 0,
+        debug: int = 0,
+        lazy_discover: bool = True,
+    ) -> None:
+        super().__init__(_MAPPING, ip, token, start_id, debug, lazy_discover)
+
+    @command(
+        default_output=format_output(
+            "",
+            "First Switch Status: {result.switch_1_state}\n"
+            "First Switch Default State: {result.switch_1_default_state}\n"
+            "First Switch Delay: {result.switch_1_off_delay}\n"
+            "Second Switch Status: {result.switch_2_state}\n"
+            "Second Switch Default State: {result.switch_2_default_state}\n"
+            "Second Switch Delay: {result.switch_2_off_delay}\n"
+            "Interlock: {result.interlock}\n"
+            "Flex Mode: {result.flex_mode}\n"
+            "RC list: {result.rc_list}\n",
+        )
+    )
+    def status(self) -> DualControlModuleStatus:
+        """Retrieve properties"""
+        p = [
+            "switch_1_state",
+            "switch_1_default_state",
+            "switch_1_off_delay",
+            "switch_2_state",
+            "switch_2_default_state",
+            "switch_2_off_delay",
+            "interlock",
+            "flex_mode",
+            "rc_list",
+        ]
+        properties = [
+            {"did": k, **v}
+            for k, v in filter(lambda item: item[0] in p, _MAPPING.items())
```

Comment:
Please add a note for why this filtering is necessary. Also, I think you should use `get_properties_for_mapping()` here like other implementations do, did you try that already and was it not working?

Reply:
I use it because not all props from mapping are readable.
For ex. toggle accept only `write` opperation and it throw exception during get status.

Reply:
I thought so. Please add a note about it, so that it doesn't get forgotten. I hope the mappings can be improved in the future to avoid requesting write-only siid/piid combinations but it's not a priority of mine at the moment.